src/components/cc-pricing-estimation/cc-pricing-estimation.smart.js

```diff
@@ -0,0 +1,76 @@
+import { fetchPriceSystem } from '../../lib/api-helpers.js';
+import { defineSmartComponent } from '../../lib/define-smart-component.js';
+import { formatEstimationPrices } from '../../lib/product.js';
+import '../cc-smart-container/cc-smart-container.js';
+import './cc-pricing-estimation.js';
+
+/**
+ * @typedef {import('./cc-pricing-estimation.js').CcPricingEstimation} CcPricingEstimation
+ * @typedef {import('./cc-pricing-estimation.types.js').PricingEstimationStateLoaded} PricingEstimationStateLoaded
+ * @typedef {import('../../lib/send-to-api.types.js').ApiConfig} ApiConfig
+ */
+
+defineSmartComponent({
+  selector: 'cc-pricing-estimation',
+  params: {
+    apiConfig: { type: Object, optional: true },
```

Comment:
The issue about smart components not being able to handle cases where all params are optional is tracked but you're right, we should avoid making all params optional until it's fixed.

As for which params we make optional: 
For most of the pricing components, it kinda makes sense to set `zoneId` and `currency` optional because they can be used in a context with no `cc-pricing-header` (product pages on the website + the heptapod website). 

Without `cc-pricing-header` & `cc-pricing-estimation` you cannot change the zone or the currency so it's great to have default values and not force people to set these.

I think I'll make `apiConfig` mandatory again and update the issue so that we change it to optional once the smart can handle that.